programs/mp-sol-restaking/src/instructions/admin/configure_main_vault.rs

```diff
@@ -1,52 +1,104 @@
 use crate::{error::ErrorCode, state::MainVaultState, MAX_PERFORMANCE_FEE_BP, MAX_WITHDRAW_FEE_BP};
 use anchor_lang::prelude::*;
+use anchor_spl::token::{Mint, TokenAccount};
 
-#[derive(AnchorSerialize, AnchorDeserialize, Debug)]
-pub struct ConfigureMainVaultValues {
-    unstake_ticket_waiting_hours: Option<u16>,
-    withdraw_fee_bp: Option<u16>,
-    performance_fee_bp: Option<u16>,
-    treasury_mpsol_account: Option<Pubkey>,
-    new_admin_pubkey: Option<Pubkey>,
+#[derive(Accounts)]
+pub struct ConfigureTreasuryAccount<'info> {
+    #[account()]
+    pub admin: Signer<'info>,
+
+    #[account(mut, has_one = admin, has_one = mpsol_mint)]
+    pub main_state: Account<'info, MainVaultState>,
+
+    #[account()]
+    pub mpsol_mint: Box<Account<'info, Mint>>,
+
+    #[account(token::mint = mpsol_mint)]
```

Comment:
Please make it optional so we have a way to set it back to None

Reply:
does Squads allows you to set optional arguments?

Reply:
will add separate clear instructions